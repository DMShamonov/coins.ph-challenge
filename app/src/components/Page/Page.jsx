// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PagePropsType } from './types.js.flow';

export default function Page({
  className,
  children,
}: PagePropsType): React.Node {
  return (
    <section className={classnames('page', className)}>
      {children}
    </section>
  );
}
