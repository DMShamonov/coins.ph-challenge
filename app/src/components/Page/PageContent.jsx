// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PageContentPropsType } from './types.js.flow';

export default function PageContent({
  className,
  children,
}: PageContentPropsType): React.Node {
  return (
    <div className={classnames('page__content', className)}>
      {children}
    </div>
  );
}
