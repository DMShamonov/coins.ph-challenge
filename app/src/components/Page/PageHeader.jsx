// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PageHeaderPropsType } from './types.js.flow';

export default function PageHeader({
  className,
  children,
}: PageHeaderPropsType): React.Node {
  return (
    <header className={classnames('page__header', className)}>
      {children}
    </header>
  );
}
