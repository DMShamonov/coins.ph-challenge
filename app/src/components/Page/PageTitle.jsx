// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PageTitlePropsType } from './types.js.flow';

export default function PageTitle({
  className,
  children,
}: PageTitlePropsType): React.Node {
  return (
    <h1 className={classnames('page__title', className)}>
      {children}
    </h1>
  );
}
