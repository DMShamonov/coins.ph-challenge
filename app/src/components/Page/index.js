export { default as Page } from './Page';
export { default as PageHeader } from './PageHeader';
export { default as PageContent } from './PageContent';
export { default as PageTitle } from './PageTitle';
