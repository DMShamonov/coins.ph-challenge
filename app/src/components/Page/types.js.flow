// @flow

import type { Node } from 'react';

export type PagePropsType = {|
  className?: string,
  children: Node,
|};

export type PageHeaderPropsType = {|
  className?: string,
  children: Node,
|};

export type PageContentPropsType = {|
  className?: string,
  children: Node,
|};

export type PageTitlePropsType = {|
  className?: string,
  children: Node,
|};
