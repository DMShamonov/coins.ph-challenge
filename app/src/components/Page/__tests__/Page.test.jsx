import React from 'react';
import { shallow } from 'enzyme';

import Page from '../Page';

describe('`Page` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Page>__CONTENT__</Page>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Page className="__CLASSNAME__">__CONTENT__</Page>)).toMatchSnapshot();
  });
});
