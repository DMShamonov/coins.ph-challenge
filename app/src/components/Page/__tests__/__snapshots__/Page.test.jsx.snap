// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Page\` component should render correctly with \`className\` prop 1`] = `
<section
  className="page __CLASSNAME__"
>
  __CONTENT__
</section>
`;

exports[`\`Page\` component should render correctly with default props 1`] = `
<section
  className="page"
>
  __CONTENT__
</section>
`;
